sonar-orchestrator-junit5/src/main/java/com/sonar/orchestrator/junit5/OrchestratorExtension.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Orchestrator - JUnit 5
+ * Copyright (C) 2011-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonar.orchestrator.junit5;
+
+import com.sonar.orchestrator.Orchestrator;
+import com.sonar.orchestrator.config.Configuration;
+import com.sonar.orchestrator.container.SonarDistribution;
+import com.sonar.orchestrator.server.StartupLogWatcher;
+import com.sonar.orchestrator.version.Version;
+import java.lang.reflect.AnnotatedElement;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.jetbrains.annotations.Nullable;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ConditionEvaluationResult;
+import org.junit.jupiter.api.extension.ExecutionCondition;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.platform.commons.util.AnnotationUtils;
+
+/**
+ * To be used as a JUnit 5 {@link org.junit.jupiter.api.extension.Extension}. For example:
+ * <pre>{@code
+ * public class MyTest {
+ *
+ *   @RegisterExtension
+ *   static OrchestratorExtension ORCHESTRATOR = OrchestratorExtension.builderEnv()
+ *     .setSonarVersion("LATEST_RELEASE")
+ *     .addPlugin(MavenLocation.of("org.sonarsource.html", "sonar-html-plugin", "DEV"))
+ *     .restoreProfileAtStartup(FileLocation.ofClasspath("/profile-jsp.xml"))
+ *     .build();
+ * }
+ * }</pre>
+ */
+public class OrchestratorExtension extends Orchestrator implements BeforeAllCallback, AfterAllCallback, ExecutionCondition {
```

Comment:
I was a bit surprised that the extension extends the Orchestrator class, but I guess that works as long as there is one instance of Orchestrator for one extension

Reply:
This allows you to call all Orchestrator methods directly on the extension, without having to wrap.